### PR TITLE
feat: Implement Issue #1779: add rollout.Spec.Strategy.Canary.MinPodsPerReplicaSet

### PR DESCRIPTION
**Pull request by ssanders1449**
Implements issue https://github.com/argoproj/argo-rollouts/issues/1779
See https://github.com/argoproj/argo-rollouts/issues/1779 for Use Cases

This is option 1 from https://github.com/argoproj/argo-rollouts/issues/1779#issuecomment-1323867218

Signed-off-by: Shlomo Sanders <ssanders@synamedia.com>

### CODE REVIEW

**Comment by ssanders1449** [comment]
@zachaller 
Please take a look at this PR

File: pkg/apis/rollouts/v1alpha1/types.go [hunk 1]
```diff
@@ -310,6 +310,9 @@ type CanaryStrategy struct {
 	DynamicStableScale bool `json:"dynamicStableScale,omitempty" protobuf:"varint,14,opt,name=dynamicStableScale"`
 	// PingPongSpec holds the ping and pong services
 	PingPong *PingPongSpec `json:"pingPong,omitempty" protobuf:"varint,15,opt,name=pingPong"`
+	// Assuming the desired number of pods in a stable or canary ReplicaSet is not zero, then make sure it is at least
+	// MinPodsPerRS for High Availability. Only applicable for TrafficRoutedCanary
+	MinPodsPerRS *int32 `json:"minPodsPerRS,omitempty" protobuf:"varint,16,opt,name=minPodsPerRS"`
```
Comment (zachaller) [inline pkg/apis/rollouts/v1alpha1/types.go:315]: Can we use ReplicaSet instead of RS

Comment (zachaller) [reply pkg/apis/rollouts/v1alpha1/types.go:315]: MinPodsPerReplicaSet

Comment (ssanders1449) [reply pkg/apis/rollouts/v1alpha1/types.go]: done

**Comment by zachaller** [comment]
Sorry should have cought this in my first look as well. Can we get some docs added to at least the specs page https://argoproj.github.io/argo-rollouts/features/specification/

**Comment by zachaller** [comment]
I want to get this in 1.4 can you please resolve conflicts?

**Comment by zachaller** [comment]
actually nm, i see you enabled maintainers to pushed it was just a codegen issue so I was able to fix it :D

**Review by zachaller — APPROVED** [review]

**Pull request merged**